pkg/bootstrapper/bootstrapper.go

```diff
@@ -492,6 +533,164 @@ func (wmcb *winNodeBootstrapper) InitializeKubelet() error {
+	cniPathInfo, err := os.Stat(wmcb.cniPath)
+	if err != nil {
+		return fmt.Errorf("error accessing CNI path %s: %v", wmcb.cniPath, err)
+	}
+	if !cniPathInfo.IsDir() {
+		return fmt.Errorf("CNI path cannot be a file")
+	}
+
+	// Check if there are any issues accessing the CNI configuration file. We don't want to proceed on any error as it
+	// could cause issues further down the line when copying the files.
+	cniConfigInfo, err := os.Stat(wmcb.cniConfig)
+	if err != nil {
+		return fmt.Errorf("error accessing CNI config %s: %v", wmcb.cniConfig, err)
+	}
+	if cniConfigInfo.IsDir() {
+		return fmt.Errorf("CNI config cannot be a directory")
+	}
+
+	return nil
+}
+
+// copyCNIFiles() copies the CNI binaries and config to the installation directory
+func (wmcb *winNodeBootstrapper) copyCNIFiles() error {
+	// Example: C:\k\ + cni
+	// cniDir := filepath.Join(wmcb.installDir, cniDirName)
+	files, err := ioutil.ReadDir(wmcb.cniPath)
+	if err != nil {
+		return fmt.Errorf("error reading CNI path %s: %v", wmcb.cniPath, err)
+	}
+	if len(files) == 0 {
+		return fmt.Errorf("no files present in CNI path %s", wmcb.cniPath)
+	}
+
+	// Copy the CNI binaries from the input CNI path to the CNI installation directory
+	for _, file := range files {
+		// Ignore directories for now. If we find that there are CNI packages with nested directories, we can update
+		// this to loop to be recursive.
+		if file.IsDir() {
+			continue
+		}
+
+		// C:\source\cni\filename
+		src := filepath.Join(wmcb.cniPath, file.Name())
+		// C:\k\cni\filename
+		dest := filepath.Join(wmcb.cniInstallDir, file.Name())
+		if err = copyFile(src, dest); err != nil {
+			return fmt.Errorf("error copying %s --> %s: %v", src, dest, err)
+		}
+	}
+
+	// Copy the CNI config to the CNI configuration directory. Example: C:\k\cni\config\cni.conf
+	cniConfigDest := filepath.Join(wmcb.cniConfigInstallPath, filepath.Base(wmcb.cniConfig))
+	if err = copyFile(wmcb.cniConfig, cniConfigDest); err != nil {
+		return fmt.Errorf("error copying CNI config %s --> %s: %v", wmcb.cniConfig, cniConfigDest, err)
+	}
+	return nil
+}
+
+// ensureCNIDirIsPresent ensures that CNI parent and child directories are present on the is present on the system.
+func (wmcb *winNodeBootstrapper) ensureCNIDirIsPresent() error {
+	// By checking for the config directory, we can ensure both parent and child directories are present
+	configDir := filepath.Join(wmcb.installDir, cniConfigDirName)
+	if _, err := os.Stat(configDir); err != nil {
+		if os.IsNotExist(err) {
+			// 0700 == Only user has access
+			if err = os.MkdirAll(configDir, 0700); err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// updateKubeletArgs updates the kubelet command with the given option and value. search is used to find the the option
+// within the args. Note that if the option exists, it will be overwritten.
+func updateKubeletArgs(kubeletCmd *string, option, value string, search *regexp.Regexp) {
```

Comment:
Does `search` need to be passed in? 
It seems like all the options will be satisfied by the regex format:
```option + `=(\S*)`)```

The regex could then be compiled within the function

Reply:
>It seems like all the options will be satisfied by the regex format:
option + `=(\S*)`)

`networkPluginRegex` does not need then. I would like to be as accurate as possible with them to avoid issues.

>The regex could then be compiled within the function

It is preferable to have regexes to be compiled just once during initialization and not every time the function is called.

Reply:
:+1: